File: Java/Cristian Maldonado/Grafos/Domino_DFS.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class Domino_DFS {

    public static void main(String[] args) {
        int dominos, caminos, consultas, desde, hasta, origen;
        dominos = sc.nextInt();
        caminos = sc.nextInt();
        consultas = sc.nextInt();
        for (int i = 0; i < dominos; i++) {
            ady.add(new ArrayList<>());
        }
        while (caminos > 0) {
            desde = sc.nextInt();
            hasta = sc.nextInt();
            ady.get(desde).add(hasta);
            caminos--;
        }
        while (consultas > 0) {
            origen = sc.nextInt();
            total = 0;
            Arrays.fill(visitados, false);
            DFS(origen);
            System.out.println("Total dominos tumbados: " + total);
            consultas--;
        }
    }

    static final int MAX = 10001;
    static ArrayList<ArrayList<Integer>> ady = new ArrayList<ArrayList<Integer>>();
    static Scanner sc = new Scanner(System.in);
    static int total;
    static boolean visitados[] = new boolean[MAX];

    static void DFS(int u) {
        total++;
        visitados[u] = true;
        for (int v = 0; v < ady.get(u).size(); v++) {
            if (!visitados[ady.get(u).get(v)]) {
                DFS(ady.get(u).get(v));
            }
        }
    }
}

File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Grafos/Domino_DFS.cpp
#include <bits/stdc++.h>
#include <cstdlib>
#define MAX_V 10001

using namespace std;

bool visitados[MAX_V];
int total = 0;
vector<vector<int>> matriz(MAX_V);

void DFS(int u){
    total++;
    visitados[u] = true;
    for(int v = 0; v < matriz[u].size(); v++){
        if(!visitados[matriz[u][v]]){
            DFS(matriz[u][v]);
        }
    }
}

int main()
{
    int dominos, caminos, consultas, desde, hasta, origen;
    scanf("%i %i %i", &dominos, &caminos, &consultas);
    while(caminos > 0){
        scanf("%i %i", &desde, &hasta);
        matriz[desde].push_back(hasta);
        caminos--;
    }
    while(consultas > 0){
        scanf("%i", &origen);
        total = 0;
        memset(visitados, false, sizeof (visitados));
        DFS(origen);
        printf("Total dominos tumbados: %i \n", total);
        consultas--;
    }
    return 0;
}
